src/protocols/secure_channel/CASESession.cpp

```diff
@@ -1045,36 +982,37 @@ CHIP_ERROR CASESession::HandleSigmaR3(System::PacketBufferHandle & msg)
     err = decryptedDataTlvReader.EnterContainer(containerType);
     SuccessOrExit(err);
 
-    err = decryptedDataTlvReader.FindElementWithTag(CASETLVTag::kNOC, suppTlvReader);
+    err = decryptedDataTlvReader.FindElementWithTag(TLV::ContextTag(1), suppTlvReader);
     SuccessOrExit(err);
     VerifyOrExit(suppTlvReader.GetType() == TLV::kTLVType_ByteString, err = CHIP_ERROR_WRONG_TLV_TYPE);
     responderOpCertLen = static_cast<uint16_t>(suppTlvReader.GetLength());
     err                = suppTlvReader.GetBytes(responderOpCert, responderOpCertLen);
     SuccessOrExit(err);
 
-    // Step 3
+    // Step 5
     // Validate initiator identity located in msg->Start()
     // Constructing responder identity
-    err = Validate_and_RetrieveResponderID(responderOpCert, responderOpCertLen, remoteCredential);
+    err = Validate_and_RetrieveResponderID(ByteSpan(responderOpCert, responderOpCertLen), remoteCredential);
     SuccessOrExit(err);
 
-    // Step 4
+    // Step 4 - Construct SigmaR3 TBS Data
     msg_r3_signed_len =
         static_cast<uint16_t>(sizeof(uint16_t) + responderOpCertLen + kP256_PublicKey_Length * 2 + sizeof(uint64_t) * 3);
 
     VerifyOrExit(msg_R3_Signed.Alloc(msg_r3_signed_len), err = CHIP_ERROR_NO_MEMORY);
 
-    err = ConstructTBS3Data(responderOpCert, responderOpCertLen, msg_R3_Signed.Get(), msg_r3_signed_len);
+    err = ConstructTBS3Data(ByteSpan(responderOpCert, responderOpCertLen), msg_R3_Signed.Get(), msg_r3_signed_len);
     SuccessOrExit(err);
 
-    err = decryptedDataTlvReader.FindElementWithTag(CASETLVTag::kSignature, suppTlvReader);
+    err = decryptedDataTlvReader.FindElementWithTag(TLV::ContextTag(2), suppTlvReader);
     SuccessOrExit(err);
     VerifyOrExit(suppTlvReader.GetType() == TLV::kTLVType_ByteString, err = CHIP_ERROR_WRONG_TLV_TYPE);
     VerifyOrExit(sigmaR3SignedData.Capacity() >= suppTlvReader.GetLength(), err = CHIP_ERROR_INVALID_TLV_ELEMENT);
```

Comment:
A better name for sigmaR3SignedData is `TBSData3Signature`. Tag 2 contains a signature, not "signed" data.

Reply:
Updated.